web/v1/redshift/sql-runner/sql/custom/01-indicative-export/02-indicative-user-stitching.sql

```diff
@@ -0,0 +1,33 @@
+-- Create user mapping tables for this run
+-- Also update the output table in preparation for joining to `indicative_export_staged` table
+
+CREATE TABLE {{.scratch_schema}}.user_stitching_this_run{{.entropy}} AS (
+
+  WITH users_this_run AS (
+    SELECT
+      domain_userid,
+      MIN(user_id) as user_id,
+      MIN(derived_tstamp) AS first_seen_tstamp
+    FROM {{.scratch_schema}}.events_staged{{.entropy}}
+    GROUP BY 1
+  )
+  SELECT
+    utr.domain_userid,
+    COALESCE(us.user_id, utr.user_id) AS user_id
+    COALESCE(us.first_seen_tstamp, utr.first_seen_tstamp) AS first_seen_tstamp
+  FROM users_this_run AS utr
+  LEFT JOIN {{.output_schema}}.user_stitching{{.entropy}} us
+  ON utr.domain_userid = us.domain_userid
+  AND us.first_seen_tstamp < utr.first_seen_tstamp
+);
+
+DELETE FROM {{.output_schema}}.user_stitching {{.entropy}}
+WHERE domain_userid IN (SELECT domain_userid FROM {{.scratch_schema}}.user_stitching_this_run{{.entropy}}
+AND first_seen_tstamp > (SELECT MIN(first_seen_tstamp) FROM {{.scratch_schema}}.user_stitching_this_run{{.entropy}});
+
+UPDATE {{.output_schema}}.indicative_export set alias_user_id = COALESCE(userfields)
+FROM {{.scratch_schema}}.user_stitching_this_run{{.entropy}}
```

Comment:
General point on user stitching: Did we find out the answer to Colm's question:
>Quick question - once we identify a user with a new user_id, do we need to attribute all events from then on or do we need to attribute all events for that user?

This is going to influence the design of this model

Reply:
Worth establishing this. If not specified we can make our own call.

Reply:
Based on this from Indicative I would think it means match _all_ events to that user_id? 

* If there is no known user id, the anonymous user id is used to search for a related known user id. If a match is found, that known user id is used.